File: api/admin/exception_log/v1/exception_log.go
package v1

import "github.com/gogf/gf/v2/frame/g"

type ExceptionLogListReq struct {
	g.Meta    `path:"/exception-log/list" method:"get" tags:"异常日志" summary:"异常日志列表"`
	Page      int    `json:"page"`
	Size      int    `json:"size"`
	TraceId   string `json:"trace_id"`
	Path      string `json:"path"`
	StartTime string `json:"start_time"`
	EndTime   string `json:"end_time"`
}

type ExceptionLogListRes struct {
	List  []*ExceptionLogItem `json:"list"`
	Total int                 `json:"total"`
}

type ExceptionLogItem struct {
	Id         int64  `json:"id"`
	TraceId    string `json:"trace_id"`
	Path       string `json:"path"`
	Method     string `json:"method"`
	ErrorMsg   string `json:"error_msg"`
	Stack      string `json:"stack"`
	UserId     int64  `json:"user_id"`
	Ip         string `json:"ip"`
	CreateTime string `json:"create_time"`
}

File: manage/src/api/exceptionLog.ts
import request from './request'

export interface ExceptionLogItem {
  id: number
  trace_id: string
  path: string
  method: string
  error_msg: string
  stack: string
  user_id: number
  ip: string
  create_time: string
}

export function fetchExceptionLogList(params: {
  page?: number
  size?: number
  trace_id?: string
  path?: string
  start_time?: string
  end_time?: string
}) {
  return request.get<unknown, { data?: { list?: ExceptionLogItem[]; total?: number } }>(
    '/admin/exception-log/list',
    {
      params: {
        page: params.page ?? 1,
        size: params.size ?? 10,
        trace_id: params.trace_id,
        path: params.path,
        start_time: params.start_time,
        end_time: params.end_time,
      },
    }
  )
}
